model/task/task.go

```diff
@@ -2645,19 +2651,44 @@ func (t *Task) MarkUnscheduled() error {
 
 }
 
-// MarkUnattainableDependency updates the unattainable field for the dependency in the task's dependency list,
-// and logs if the task is newly blocked.
-func (t *Task) MarkUnattainableDependency(ctx context.Context, dependencyId string, unattainable bool) error {
-	wasBlocked := t.Blocked()
-	if err := t.updateAllMatchingDependenciesForTask(ctx, dependencyId, unattainable); err != nil {
-		return errors.Wrapf(err, "updating matching dependencies for task '%s'", t.Id)
+// MarkAllForUnattainableDependency updates many tasks (taskIDs) to mark
+// the dependency (dependencyID) as attainable or not. If marking a dependency
+// unattainable, it creates an event log for each newly blocked task. This
+// returns all the tasks after the update.
+func MarkAllForUnattainableDependency(ctx context.Context, tasks []Task, dependencyID string, unattainable bool) ([]Task, error) {
+	if len(tasks) == 0 {
+		return tasks, nil
+	}
+
+	taskIDs := make([]string, 0, len(tasks))
+	newlyBlockedTaskData := make([]event.TaskBlockedData, 0, len(tasks))
+	for _, t := range tasks {
+		if !t.Blocked() && unattainable && !t.OverrideDependencies {
```

Comment:
why do we only want to log in this case? If we're overriding dependencies, should we also not be adding to taskIDs?

Reply:
I don't really know why overriding dependencies matters, but this check is equivalent to the old one, so I just kept the logic the same. It should still log task blocked under the same conditions as before.